testing/unittests/metrics/MetricFrameworkTests.cpp

```diff
@@ -209,13 +210,62 @@ class MetricFrameworkTests : public CppUnit::TestFixture
             ScopedGaugeDecrementer gaugeDecrementer(*pGauge, 1);
         }
         CPPUNIT_ASSERT_EQUAL(0, static_cast<int>(pGauge->queryValue()));
+    }
+
+    void Test_metric_meta_data()
+    {
+
+        //
+        // Test that two metrics with the same base name, but differing meta data are added w/o error. Note code shows passing meta data as an externally
+        // constructed vector and as a vector constructed in place.
+        MetricMetaData metaData1{{"key1", "value1"}};
+        std::shared_ptr<CounterMetric> pCounter1 = std::make_shared<CounterMetric>("requests.completed", "description", SMeasureCount, metaData1);
+        std::shared_ptr<CounterMetric> pCounter2 = std::make_shared<CounterMetric>("requests.completed", "description", SMeasureCount,
+                                                                                   MetricMetaData{{"key1", "value2"}});
+
+        frameworkTestManager.addMetric(pCounter1);
+
+        try
+        {
+            bool rc = frameworkTestManager.addMetric(pCounter2);
+            CPPUNIT_ASSERT(rc);
+        }
+        catch (...)
+        {
+            CPPUNIT_ASSERT(false);  // force an assert since no exception should be thrown
+        }
+
+
+        //
+        // Test that two metrics with the same base name and meta data fail adding. Again note both methods for passing meta data
+        MetricMetaData metaData2{{"key1", "value1"}};
+        std::shared_ptr<CounterMetric> pCounter3 = std::make_shared<CounterMetric>("requests.queued", "description", SMeasureCount, metaData2);
+        std::shared_ptr<CounterMetric> pCounter4 = std::make_shared<CounterMetric>("requests.queued", "description", SMeasureCount,
+                                                                                   MetricMetaData{{"key1", "value1"}});
+
+        frameworkTestManager.addMetric(pCounter3);
+
+        bool exceptionThrown = false;
```

Comment:
Simpler and more consistent would be:

```
        bool success = true;
        try
        {
            // Expect a return of false if not debug
            success = frameworkTestManager.addMetric(pCounter4);
        }
        catch (IException * e)
        {
            success = false;   // this is expected if built in debug mode
            e->Release();
        }

        CPPUNIT_ASSERT(!success);

```
It would also be worth adding another test:

`pCounter3.reset();
`
and then checking that adding the metric now succeeds.

Reply:
Added a new test.